packages/modul-components/src/components/responsive-table/responsive-table.ts

```diff
@@ -177,7 +177,26 @@ export class MResponsiveTable extends ModulVue {
 
     @Watch('headRows', { immediate: true, deep: true })
     public onHeadRowsChange(headRows: MTableHeadRows): void {
-        this.headRowsFilterAndSort = headRows;
+        if (headRows && !(this.columns && this.columns.length)) {
+            this.headRowsFilterAndSort = headRows;
+        }
+    }
+
+    @Watch('columns', { immediate: true, deep: true })
+    public onColumnsChange(columns: MTableColumn[]): void {
+        if (columns && columns.length) {
```

Comment:
💡  pas besoin de columns... aussi je ferais un early return si !columns.length .. le code serait moins indenté

Reply:
this.columns.length !== 0